interfaces/builtin/ofono.go

```diff
@@ -0,0 +1,314 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+const ofonoPermanentSlotAppArmor = `
+# Description: Allow operating as the ofono service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# ofono puts ppp on top of the tun device
+/dev/net/tun rw,
+# to create ppp network interfaces
+capability net_admin,
+
+# To check present devices
+/run/udev/data/+usb:* r,
+/run/udev/data/+pci:* r,
+/run/udev/data/c* r,
+/run/udev/data/n* r,
```

Comment:
These aren't quite as fine-grained as I was hoping. It seems from the udev rules that only a handful of devices are known by ofono-- is that true? If so, can you determine the more fine-grained access? If you need help, simply remove these, run in devmode and paste the /run/udev/data denials and we can get more fine-grained.